### PR TITLE
fix: Cut release only when all tests passed

### PR DESCRIPTION
**Pull request by goruha**
## what
* Run release job only if all tests passed

## why
* Do not release failed binaries

## references
* https://github.com/cloudposse/atmos/actions/runs/11307359580

## Summary by CodeRabbit

- **New Features**
	- Reintroduced an automated release job that enhances the deployment process, triggered on push events.
  
- **Bug Fixes**
	- Corrected indentation and structure in the Docker job for improved workflow execution.

### CODE REVIEW

**Review by kevcube — APPROVED** [review]
seems like a good idea

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -329,3 +320,13 @@ jobs:
...
           fail_on_error: true
+
+  release:
+    needs: [test, lint, mock, k3s, localstack, docker]
+    if: github.event_name == 'push'
+    uses: cloudposse/.github/.github/workflows/shared-go-auto-release.yml@main
+    with:
+      publish: false
+      format: binary
+    secrets: inherit
```
Comment (kevcube) [inline .github/workflows/test.yml:331]: i don't agree that `release` should be in a workflow called `test.yml` with `name: Tests`

but it was already here to begin with so i won't require it to move out

Comment (osterman) [reply .github/workflows/test.yml:331]: @goruha I think @kevcube brings up a good point naming wise. I would not think any release happens from a workflow file called test.

**Review by bateller — APPROVED** [review]

**Comment by kevcube** [comment]
I also feel that there isn't a huge need to require tests to pass before releases happen - instead, tests should be required for any PR merge. If tests are deterministic, which I expect them to be, then all code in `main` should pass tests if every PR must pass tests, and then releases can happen more promptly post-merge. But that's outside of scope for what this PR needs to fix, just voicing my opinion on it.

**Comment by goruha** [comment]
@kevcube I'd agree with you skipping tests on main branch runs if we would require `base update` for all PRs. 
The const of that is to slow down PR merging throughput. 
Also, as we see, sometimes tests on the main branch fail so the check is not useless.

Could you pls approve this PR?

**Pull request merged**